exonum/src/sandbox/sandbox.rs

```diff
@@ -632,6 +635,33 @@ impl Sandbox {
     fn node_secret_key(&self) -> SecretKey {
         self.node_state().consensus_secret_key().clone()
     }
+
+    fn add_peer_to_connect_list(&self, addr: SocketAddr, public_key: PublicKey) {
+        let config = {
+            let inner = &self.inner.borrow_mut();
+            let state = &inner.handler.state;
+            let mut config = state.config().clone();
+            config.validator_keys.push(ValidatorKeys {
+                consensus_key: public_key.clone(),
```

Comment:
I don't understand this code at all. Consensus and service keys should never coincide.